### PR TITLE
Adding package READMEs and updating package metadata

### PR DESCRIPTION
**Pull request by mathewc**
All Microsoft developed packages should include readmes - these WebJobs SDK packages currently don't. Open to feedback on readme text/samples, as well as package descriptions.

### CODE REVIEW

File: src/Microsoft.Azure.WebJobs.Extensions.Rpc/README.md [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+## Commonly used types
+
+- `WebJobsExtensionBuilderRpcExtensions`
+
+## Example usage
+
+The below example demonstrates how an extension can register a custom gRPC extension.
+
+``` CSharp
+public static IWebJobsBuilder AddServiceBus(this IWebJobsBuilder builder, Action<ServiceBusOptions> configure)
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Extensions.Rpc/README.md]: ServiceBus is the only example I know of doing this, so chose a sample based on their usage.

File: src/Microsoft.Azure.WebJobs.Extensions.Rpc/WebJobs.Extensions.Rpc.csproj [hunk 2]
```diff
@@ -5,9 +5,9 @@
     <TargetFramework>net6.0</TargetFramework>
     <LangVersion>latest</LangVersion>
     <PackageId>Microsoft.Azure.WebJobs.Extensions.Rpc</PackageId>
+    <Description>This package provides RPC capabilities to the WebJobs SDK, allowing extensions to communicate between the host and worker via RPC. For more information, please visit https://learn.microsoft.com/en-us/azure/app-service/webjobs-sdk-get-started.</Description>
     <AssemblyName>Microsoft.Azure.WebJobs.Extensions.Rpc</AssemblyName>
     <RootNamespace>Microsoft.Azure.WebJobs.Extensions.Rpc</RootNamespace>
-    <Description>This package contains the runtime assemblies for Microsoft.Azure.WebJobs.Extensions.Rpc. For more information, please visit http://go.microsoft.com/fwlink/?LinkID=320971</Description>
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Extensions.Rpc/WebJobs.Extensions.Rpc.csproj:10]: As part of updating the Descriptions, I'm also updating the "more information" URI to point to our official WebJobs SDK tutorial. Existing description points to wiki.

File: src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/WebJobs.Logging.ApplicationInsights.csproj [hunk 3]
```diff
@@ -50,7 +54,7 @@
     without all of the others in this solution. If changes to WebJobs.Host are needed here, re-introduce the
     project reference. 
     -->
-    <PackageReference Include="Microsoft.Azure.WebJobs" Version="3.0.33" />
+    <PackageReference Include="Microsoft.Azure.WebJobs" Version="3.0.39" />
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/WebJobs.Logging.ApplicationInsights.csproj:57]: This was referencing a really old version, so I'm bumping this to the latest released.

File: src/Microsoft.Azure.WebJobs.Logging/README.md [hunk 4]
```diff
@@ -0,0 +1,9 @@
+﻿# Microsoft.Azure.WebJobs.Host.Logging
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Logging/README.md:1]: This Logging package doesn't seem to be used by anyone anymore. It's the old Cosmos Table logging functionality.

File: src/Microsoft.Azure.WebJobs.Host/WebJobs.Host.Sources.csproj [hunk 5]
```diff
@@ -16,6 +17,10 @@
     <GeneratePackageOnBuild>true</GeneratePackageOnBuild>
   </PropertyGroup>
 
+  <ItemGroup>
+    <Content Include="README.md" Pack="true" PackagePath="/" />
```
Comment (mathewc) [inline src/Microsoft.Azure.WebJobs.Host/WebJobs.Host.Sources.csproj:21]: Choosing to use the same readme for both the WebJobs.Host and WebJobs.Host.Sources packages.

File: src/Microsoft.Azure.WebJobs.Extensions.Rpc/README.md (hunk 1 above)
Comment (jviau) [reply src/Microsoft.Azure.WebJobs.Extensions.Rpc/README.md]: I recommend using placeholder names, like `AddMyExtension`, `AddExtension<MyExtensionConfigProvider>`, `MapWorkerGrpcService<MyGrpcService>`

File: src/Microsoft.Azure.WebJobs.Host/README.md [hunk 6]
```diff
@@ -0,0 +1,45 @@
...
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+class Program
+{
+    public static async Task Main(string[] args)
+    {
+        var builder = new HostBuilder()
```
Comment (jviau) [inline src/Microsoft.Azure.WebJobs.Host/README.md]: recommend switching this to:

``` csharp
var builder = Host.CreateDefaultBuilder(args)
    .ConfigureWebJobs(b =>
    {
        b.AddAzureStorageCoreServices();
        b.AddAzureStorageQueues();
    });

using var host = builder.Build();
await host.RunAsync();
```

File: src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/README.md [hunk 7]
```diff
@@ -0,0 +1,49 @@
...
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.DependencyInjection;
+using Microsoft.Extensions.Hosting;
+using Microsoft.Extensions.Logging;
+
+class Program
+{
+    public static async Task Main(string[] args)
+    {
+        var builder = new HostBuilder()
```
Comment (jviau) [inline src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/README.md]: Recommend same builder changes here.

File: src/Microsoft.Azure.WebJobs.Logging/WebJobs.Logging.csproj [hunk 8]
```diff
@@ -19,10 +19,14 @@
     <TreatWarningsAsErrors>true</TreatWarningsAsErrors>
     <WarningsAsErrors />
   </PropertyGroup>
-
+
```
Comment (jviau) [inline src/Microsoft.Azure.WebJobs.Logging/WebJobs.Logging.csproj:22]: nit whitespace

File: src/Microsoft.Azure.WebJobs.Host/README.md (hunk 6 above)
Comment (fabiocav) [reply src/Microsoft.Azure.WebJobs.Host/README.md]: +1

File: src/Microsoft.Azure.WebJobs.Host/WebJobs.Host.csproj [hunk 9]
```diff
@@ -5,7 +5,7 @@
     <TargetFramework>netstandard2.0</TargetFramework>
     <LangVersion>latest</LangVersion>
     <PackageId>Microsoft.Azure.WebJobs</PackageId>
-    <Description>This package contains the runtime assemblies for Microsoft.Azure.WebJobs.Host. It also adds rich diagnostics capabilities which makes it easier to monitor the WebJobs in the dashboard. For more information, please visit http://go.microsoft.com/fwlink/?LinkID=320971</Description>
+    <Description>This package contains the runtime host components of the WebJobs SDK. For more information, please visit https://learn.microsoft.com/en-us/azure/app-service/webjobs-sdk-get-started.</Description>
```
Comment (fabiocav) [inline src/Microsoft.Azure.WebJobs.Host/WebJobs.Host.csproj]: We should continue use fwlink for this. If needed, we can create one for the target.

File: src/Microsoft.Azure.WebJobs.Host/README.md [hunk 10]
```diff
@@ -0,0 +1,45 @@
+﻿# Microsoft.Azure.WebJobs.Host
+
+This package contains the runtime host components of the WebJobs SDK. For more information, please visit https://learn.microsoft.com/en-us/azure/app-service/webjobs-sdk-get-started.
```
Comment (fabiocav) [inline src/Microsoft.Azure.WebJobs.Host/README.md]: Update to use fwlink

File: src/Microsoft.Azure.WebJobs.Host.Storage/WebJobs.Host.Storage.csproj [hunk 11]
```diff
@@ -7,6 +7,8 @@
     <InformationalVersion>$(Version) Commit hash: $(CommitHash)</InformationalVersion>
     <TargetFramework>netstandard2.0</TargetFramework>
     <LangVersion>latest</LangVersion>
+    <PackageId>Microsoft.Azure.WebJobs.Host.Storage</PackageId>
+    <Description>This package contains Azure Storage based implementations of some WebJobs SDK component interfaces. For more information, please visit https://learn.microsoft.com/en-us/azure/app-service/webjobs-sdk-get-started.</Description>
```
Comment (fabiocav) [inline src/Microsoft.Azure.WebJobs.Host.Storage/WebJobs.Host.Storage.csproj]: Use fwlink

File: src/Microsoft.Azure.WebJobs.Host.Storage/README.md [hunk 12]
```diff
@@ -0,0 +1,47 @@
+﻿# Microsoft.Azure.WebJobs.Host.Storage
+
+This package contains Azure Storage based implementations of some WebJobs SDK component interfaces. For more information, please visit https://learn.microsoft.com/en-us/azure/app-service/webjobs-sdk-get-started.
```
Comment (fabiocav) [inline src/Microsoft.Azure.WebJobs.Host.Storage/README.md]: fwlink

File: src/Microsoft.Azure.WebJobs.Extensions.Rpc/WebJobs.Extensions.Rpc.csproj [hunk 13]
```diff
@@ -5,9 +5,9 @@
     <TargetFramework>net6.0</TargetFramework>
     <LangVersion>latest</LangVersion>
     <PackageId>Microsoft.Azure.WebJobs.Extensions.Rpc</PackageId>
+    <Description>This package provides RPC capabilities to the WebJobs SDK, allowing extensions to communicate between the host and worker via RPC. For more information, please visit https://learn.microsoft.com/en-us/azure/app-service/webjobs-sdk-get-started.</Description>
```
Comment (fabiocav) [inline src/Microsoft.Azure.WebJobs.Extensions.Rpc/WebJobs.Extensions.Rpc.csproj]: Continue to use fwlink here.

File: src/Microsoft.Azure.WebJobs.Extensions.Rpc/README.md [hunk 14]
```diff
@@ -0,0 +1,23 @@
+﻿# Microsoft.Azure.WebJobs.Extensions.Rpc
+
+This package provides RPC capabilities to the WebJobs SDK, allowing extensions to communicate between the host and worker via RPC. For more information, please visit https://learn.microsoft.com/en-us/azure/app-service/webjobs-sdk-get-started.
```
Comment (fabiocav) [inline src/Microsoft.Azure.WebJobs.Extensions.Rpc/README.md]: Same as the other comments for the link target.

File: src/Microsoft.Azure.WebJobs.Host/WebJobs.Host.Sources.csproj [hunk 15]
```diff
@@ -5,6 +5,7 @@
 
   <PropertyGroup>
     <PackageId>Microsoft.Azure.WebJobs.Sources</PackageId>
+    <Description>This package contains source for the runtime host components of the WebJobs SDK. For more information, please visit https://learn.microsoft.com/en-us/azure/app-service/webjobs-sdk-get-started.</Description>
```
Comment (fabiocav) [inline src/Microsoft.Azure.WebJobs.Host/WebJobs.Host.Sources.csproj]: Same as the other comments for the link.

File: src/Microsoft.Azure.WebJobs.Host/WebJobs.Host.csproj (hunk 9 above)
Comment (fabiocav) [reply src/Microsoft.Azure.WebJobs.Host/WebJobs.Host.csproj]: Applies to all links.

**Review by fabiocav — COMMENTED** [review]
Thanks again, @mathewc! Left comments, mostly related to the link targets

File: src/Microsoft.Azure.WebJobs.Host/WebJobs.Host.csproj (hunk 9 above)
Comment (mathewc) [reply src/Microsoft.Azure.WebJobs.Host/WebJobs.Host.csproj]: Created a new fwlink https://go.microsoft.com/fwlink/?linkid=2279708 and updated all usages

File: src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/README.md (hunk 7 above)
Comment (mathewc) [reply src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/README.md]: Applied some of your changes, however the logger config code needs to remain.

Comment (jviau) [reply src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/README.md]: Logger config is fine to remain. The important part to change is switching from `new HostBuilder()` to `Host.CreateDefaultBuilder(args)` to encourage best practices. `CreateDefaultBuilder(args)` will apply many defaults that users are used to. `new HostBuilder()` gives you a completely barebones builder with nothing preconfigured.

Can you make that change here as well as in src/Microsoft.Azure.WebJobs.Host.Storage/README.md?

**Comment by mathewc** [comment]
@fabiocav @jviau All feedback addressed, thanks.

**Comment by jviau** [comment]
@mathewc I re-opened a comment with additional feedback

https://github.com/Azure/azure-webjobs-sdk/pull/3083#discussion_r1669052156

File: src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/README.md (hunk 7 above)
Comment (mathewc) [reply src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/README.md]: Got it. Fixed all the places.

**Review by jviau — APPROVED** [review]

**Pull request merged**